pkg/lang/ir/compile.go

```diff
@@ -154,6 +161,49 @@ func (g Graph) Compile(uid, gid int) (llb.State, error) {
 	return finalStage, nil
 }
 
+func (g Graph) compileJulia(aptStage llb.State) (llb.State, error) {
+	g.compileJupyter()
+	builtinSystemStage := aptStage
+
+	sshStage, err := g.copySSHKey(builtinSystemStage)
+	if err != nil {
+		return llb.State{}, errors.Wrap(err, "failed to copy ssh keys")
+	}
+	diffSSHStage := llb.Diff(builtinSystemStage, sshStage, llb.WithCustomName("install ssh keys"))
+
+	shellStage, err := g.compileShell(builtinSystemStage)
+	if err != nil {
+		return llb.State{}, errors.Wrap(err, "failed to compile shell")
+	}
+	diffShellStage := llb.Diff(builtinSystemStage, shellStage, llb.WithCustomName("install shell"))
+
+	systemStage := llb.Diff(builtinSystemStage, g.compileSystemPackages(builtinSystemStage),
+		llb.WithCustomName("install system packages"))
+
+	// TODO(terrytangyuan): Support RStudio local server
+	juliaStage := llb.Diff(builtinSystemStage,
```

Comment:
This would need some refactoring to reduce code redundancy.

Reply:
Yep I think we need a base graph and extend it with different sub-struct like PythonGraph and JuliaGraph.